### ISSUE TITLE
Sound classifier accuracy with TF much less than with MXNet (Mac, CPU).

### ISSUE THREAD

**Issue by hoytak** [apple/turicreate#2688]
Using the following test code: 

```
import turicreate as tc
from os.path import basename

# Load the audio data and meta data.
data = tc.load_audio("/Users/hoytak/data/ESC-50/audio/")
meta_data = tc.SFrame.read_csv("~/data/ESC-50/meta/esc50.csv")

data['filename'] = data['path'].apply(lambda p: basename(p))
data = data.join(meta_data)

# Drop all records which are not part of the ESC-10.
data = data.filter_by('True', 'esc10')

# Make a train-test split, just use the first fold as our test set.
test_set = data.filter_by(1, 'fold')
train_set = data.filter_by(1, 'fold', exclude=True)

model = tc.sound_classifier.create(train_set, target='category', feature='audio')
```

On 5.8, 4 subsequent runs produced training accuracies 0.913, 0.923, 0.919, and 0.917.  With change 41ccb592f9e9ded4298662a697c729f2f62dfe0f, the accuracies were 0.856, 0.847, 0.863, 0.823.  In addition, the accuracy frequently went up, then down in the training.  It's likely the step size is set too large. 

Example runs: 
```
| 8                       | 0.820                   | 0.775                   | 69.238                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.838                   | 0.838                   | 69.567                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.823                   | 0.762                   | 69.883                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
```

```
+-------------------------+-------------------------+-------------------------+-------------------------+
| 8                       | 0.873                   | 0.787                   | 67.856                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 9                       | 0.826                   | 0.787                   | 68.174                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
| 10                      | 0.863                   | 0.800                   | 68.511                  |
+-------------------------+-------------------------+-------------------------+-------------------------+
```

**Comment by nickjong** [comment]
Looks like we went from this MXNet optimizer:
https://github.com/apple/turicreate/blob/7f20fa398695823062dc1e86723e30427259d2f6/src/python/turicreate/toolkits/sound_classifier/_mx_sound_classifier.py#L24
to this TF one:
https://github.com/apple/turicreate/blob/7f20fa398695823062dc1e86723e30427259d2f6/src/python/turicreate/toolkits/sound_classifier/_tf_sound_classifier.py#L81

Might require some experimentation and/or debugging...

**Comment by fareeha** [comment]
> Looks like we went from this MXNet optimizer:
> 
> https://github.com/apple/turicreate/blob/7f20fa398695823062dc1e86723e30427259d2f6/src/python/turicreate/toolkits/sound_classifier/_mx_sound_classifier.py#L24
> 
> 
> to this TF one:
> https://github.com/apple/turicreate/blob/7f20fa398695823062dc1e86723e30427259d2f6/src/python/turicreate/toolkits/sound_classifier/_tf_sound_classifier.py#L81
> 
> Might require some experimentation and/or debugging...

You're right Nick. So MXNET uses the NAG optimizer which takes `lr` and `momentum` and TF uses Adam optimizer which also does not use `momentum`. I meant to replace Adam. I believe the equivalent of NAG in TF is https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/MomentumOptimizer which will take in both `lr` and `momentum`. This may fix it.

**Comment by TobyRoseman** [comment]
The loss function may also be a bit different. In 5.8 we used `mx.gluon.loss.SoftmaxCrossEntropyLoss()`. Looks like now we are using `tensorflow.compat.v1.nn.softmax_cross_entropy_with_logits_v2`.

**Comment by syoutsey** [comment]
BRB: worth investigating for 6.0

**Comment by fareeha** [comment]
> curacies 0.913, 0.923, 0.919, and 0.917. With change [41ccb59](https://github.com/apple/turicreate/commit/41ccb592f9e9ded4298662a697c729f2f62dfe0f), the accuracies were 0.856, 0.847, 0.8

Was the Training and Validation set fixed between all 8 runs across 5.8 and 6.0?

**Comment by fareeha** [comment]
@hoytak Can you please give us more details on how you reproduced? I used your code, set `validation_set=None` to force using the same data in all runs and get the following numbers:

5.8
```
Training a custom neural network -
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.260             | 4.905             |
+-------------------+-------------------+-------------------+
| 2                 | 0.260             | 4.912             |
+-------------------+-------------------+-------------------+
| 3                 | 0.420             | 4.919             |
+-------------------+-------------------+-------------------+
| 4                 | 0.500             | 4.928             |
+-------------------+-------------------+-------------------+
| 5                 | 0.500             | 4.935             |
+-------------------+-------------------+-------------------+
| 6                 | 0.480             | 4.942             |
+-------------------+-------------------+-------------------+
| 7                 | 0.480             | 4.950             |
+-------------------+-------------------+-------------------+
| 8                 | 0.440             | 4.957             |
+-------------------+-------------------+-------------------+
| 9                 | 0.440             | 4.965             |
+-------------------+-------------------+-------------------+
| 10                | 0.420             | 4.973             |
+-------------------+-------------------+-------------------+
```

41ccb59 (6.0) still using the AdamOptimizer:

```
+-------------------+-------------------+-------------------+
| Iteration         | Training Accuracy | Elapsed Time      |
+-------------------+-------------------+-------------------+
| 1                 | 0.400             | 14.018            |
+-------------------+-------------------+-------------------+
| 2                 | 0.820             | 14.032            |
+-------------------+-------------------+-------------------+
| 3                 | 0.900             | 14.045            |
+-------------------+-------------------+-------------------+
| 4                 | 0.900             | 14.057            |
+-------------------+-------------------+-------------------+
| 5                 | 0.900             | 14.070            |
+-------------------+-------------------+-------------------+
| 6                 | 0.960             | 14.081            |
+-------------------+-------------------+-------------------+
| 7                 | 0.960             | 14.094            |
+-------------------+-------------------+-------------------+
| 8                 | 0.960             | 14.108            |
+-------------------+-------------------+-------------------+
| 9                 | 0.960             | 14.123            |
+-------------------+-------------------+-------------------+
| 10                | 0.960             | 14.136            |
+-------------------+-------------------+-------------------+
```

@nickjong it takes longer which is my only concern here ..

**Comment by shantanuchhabra** [comment]
Here are some numbers for comparison.

<img width="834" alt="image" src="https://user-images.githubusercontent.com/14062641/69469437-b5aade00-0d45-11ea-9423-5ebf96e871c7.png">

**Comment by fareeha** [comment]
> Here are some numbers for comparison.
> 
> <img alt="image" width="834" src="https://user-images.githubusercontent.com/14062641/69469437-b5aade00-0d45-11ea-9423-5ebf96e871c7.png">

Can you please share details on reproing these numbers?

**Comment by shantanuchhabra** [comment]
@fareeha followed up offline.
